google/cloud/pubsublite/integration_tests/publisher_integration_test.cc

```diff
@@ -0,0 +1,166 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsublite/admin_connection.h"
+#include "google/cloud/pubsublite/internal/location.h"
+#include "google/cloud/pubsublite/options.h"
+#include "google/cloud/pubsublite/publisher_connection.h"
+#include "google/cloud/internal/format_time_point.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/internal/populate_common_options.h"
+#include "google/cloud/internal/populate_grpc_options.h"
+#include "google/cloud/internal/random.h"
+#include "google/cloud/testing_util/integration_test.h"
+#include "google/cloud/version.h"
+#include <google/cloud/pubsublite/v1/admin.pb.h>
+#include <gmock/gmock.h>
+#include <regex>
+
+namespace google {
+namespace cloud {
+namespace pubsublite {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using google::cloud::pubsub::MessageBuilder;
+
+using google::cloud::pubsub::PublisherConnection;
+using google::cloud::pubsublite::v1::CreateTopicRequest;
+using google::cloud::pubsublite::v1::DeleteTopicRequest;
+using google::cloud::pubsublite::v1::ListTopicsRequest;
+using google::cloud::pubsublite_internal::MakeLocation;
+
+int constexpr kNumMessages = 10000;
```

Comment:
The single partition publishers batch up messages and send them out, so I want to test that they're able to successfully send batches once they reach capacity and when there are some leftover messages.

The default max number of messages per batch is 1000, and with 3 partitions per test topic, I changed `kNumMessages` to 5000 (3000 + some leftovers).

Does that work better given the reasoning?

Reply:
Sure.  The running time for this is a bit high:

```
//google/cloud/pubsublite/integration_tests:publisher_integration_test   PASSED in 8.1s
```

It would be nice if it was closer to 2 or 3 seconds.  Optional: consider changing the batch size to something smaller and dropping the number of messages.

Separately: I assume there is a timeout that triggers if the batch is not full?

Reply:
> consider changing the batch size

I set the max batch size to `250` via the options passed to the factory.

> I assume there is a timeout that triggers

Not exactly, what happens is that the single partition publisher gathers up all the individual messages and batches them up. If the first four 4 batches are full and the 5th batch isn't, there are still 5 batches that will be sent out (the 5th batch just isn't full but that's fine).